exo-go/src/app_runner/index.go

```diff
@@ -113,6 +114,19 @@ func (a *AppRunner) Shutdown(shutdownConfig types.ShutdownConfig) error {
 
 // Start runs the application and returns the process and returns an error if any
 func (a *AppRunner) Start() error {
+	watcherErr := make(chan error)
+	if err := a.watchServices(watcherErr); err != nil {
+		return err
+	}
+	go func() {
+		err := <-watcherErr
+		if err != nil {
+			if err := a.Shutdown(types.ShutdownConfig{CloseMessage: "Failed to restart"}); err != nil {
```

Comment:
Thoughts on changing "Failed to restart" to "Error watching services for changes". Its not guaranteed the error happened while trying to restart